src/dc_stock.rs

```diff
@@ -1,338 +1,114 @@
-use crate::constants::Event;
-use crate::context::Context;
-use crate::dc_contact::*;
-use crate::dc_tools::*;
-use crate::types::*;
-use crate::x::*;
-
-/* Return the string with the given ID by calling DC_EVENT_GET_STRING.
-The result must be free()'d! */
-pub unsafe fn dc_stock_str(context: &Context, id: libc::c_int) -> *mut libc::c_char {
-    return get_string(context, id, 0i32);
+/// Stock strings
+///
+/// These identify the string to return in [Context.stock_str].  The
+/// numbers must stay in sync with `deltachat.h` `DC_STR_*` constants.
+///
+/// See the `stock_*` methods on [Context] to use these.
+///
+/// [Context]: crate::context::Context
+#[derive(Debug, Clone, Copy, PartialEq, Eq, FromPrimitive, ToPrimitive)]
+#[repr(u32)]
+pub enum StockId {
+    NoMessages = 1,
```

Comment:
you could use strum, similar to how I used it

```rust
#[derive(EnumProperty, Debug)]
pub enum StockId {
	#[strum(props(default="No messages"))]
	NoMessage = 1,
}

assert_eq!("No messages", StockId.get_str("default).unwrap());
```

instead of the matching down below

Reply:
ref: https://github.com/Peternator7/strum#EnumProperty